File: spec/functional_test/fixtures/java/spark_callees/src/Application.java
package com.example;

import static spark.Spark.*;

public class Application {
    public static void main(String[] args) {
        post("/users", (req, res) -> {
            String name = req.queryParams("name");
            String saved = UserService.save(name);
            AuditLog.write(saved);
            res.status(201);
            return saved;
        });

        get("/profile", (req, res) -> {
            String built = buildProfile();
            AuditLog.write(built);
            return built;
        });

        get("/legacy", (req, res) -> {
            AuditLog.write("legacy");
            return getLegacy().toString();
        });
    }

    static String buildProfile() {
        return "profile";
    }

    static String getLegacy() {
        return "legacy";
    }
}

File: spec/functional_test/fixtures/kotlin/http4k_callees/src/main/kotlin/com/example/Application.kt
package com.example

import org.http4k.core.Method.GET
import org.http4k.core.Method.POST
import org.http4k.core.Request
import org.http4k.core.Response
import org.http4k.core.Status.Companion.OK
import org.http4k.routing.bind
import org.http4k.routing.routes

val app = routes(
    "/users" bind POST to { req: Request ->
        val name = req.query("name")
        val saved = UserService.save(name)
        AuditLog.write(saved)
        Response(OK)
    },

    "/profile" bind GET to { _: Request ->
        val built = buildProfile()
        AuditLog.write(built)
        Response(OK)
    },

    "/legacy" bind GET to { _: Request ->
        AuditLog.write("legacy")
        val out = getLegacy().toString()
        Response(OK)
    }
)

fun buildProfile(): String = "profile"
fun getLegacy(): String = "legacy"
